omega.js

```
@@ -104,28 +104,90 @@ function parse(primus, raw, res) {
     err = e;
   }
 
+  res.setHeader('Content-Type', 'application/json'); // set response content type
+
   if (
        err                              // No error..
     || 'object' !== typeof data         // Should be an object.
     || Array.isArray(data)              // A real object, not array.
     || !data.msg                        // The data we send should be defined.
   ) {
     res.statusCode = 500;
-    res.setHeader('Content-Type', 'application/json');
     return res.end('{ "ok": false, "reason": "invalid data structure" }');
   }
 
+  var callingRooms = primus.$ && primus.$.rooms && (Array.isArray(data.rooms) || typeof data.rooms === 'string');
   //
-  // Process the incoming messages in three different modes:
+  // Process the incoming messages in four different modes:
   //
+  // Rooms:  If primus-rooms is being used. Takes in rooms array or a string for
+  //         the room to broadcast to. Also an optional except array or a string of
+  //         sparkIDs to exclude.
   // Sparks: The data.sparks is an array with spark id's which we should write
   //         the data to.
   // Spark:  The data.sparks is the id of one single individual spark which
   //         should receive the data.
   // All:    Broadcast the message to every single connected spark if no
   //         `data.sparks` has been provided.
   //
-  if (Array.isArray(data.sparks)) {
+  if (callingRooms) {
+    // initialize except array
+    data.except = data.except || [];
+    if (typeof data.except === 'string') {
+      data.except = [data.except];
+    }
+    //get the sparks in the rooms
+    primus.room(data.rooms).clients(function(err, sparks) {
```

Error is not handled here.